### ISSUE TITLE
Multiple reference fixture not constructing properly with arrays

### ISSUE THREAD

**Issue by smitty260** [nelmio/alice#684]
When using the multiple reference fixture function ([url](https://github.com/nelmio/alice/blob/master/doc/relations-handling.md#multiple-references)) with a variable of an array type, it just constructs empty arrays. 

Example variable at top of class delcaration:
`private $exampleVar = [];`

Example of multiple references fixture:
 ```yaml
Some\Class\Namespace:
    Object:
        exampleVar: '@OtherObject{1..5}'
```
Currently using this workaround:
```yaml
Doctrine\Common\Collections\ArrayCollection:
  Objects:
    __construct: [ '@OtherObject{1..5}' ]
```

**Comment by theofidry** [comment]
That's a nice workaround :)

That's quite possible, I remember coming across an issue along those lines. If it has not been fixed it can be one of the following reasons:

- I was too busy with 3.x, I just made sure it's fixed in 3.x but couldn't find the time to fix it in 2.x as well
- There was an issue with fixing it: either way too complex or not possible without BC breaks

I'm unfortunately not willing to maintain 2.x anymore (PR/issues are still managed, but I'm not working actively on it), mostly because 2.x is very hard to maintain and I have far too little time (this is why 3.x, besides addressing some architectural problems of 2.x, is way more tested and I believe is way easier to maintain).

So TL:DR; either find a fix and submit a PR, or continue with your workaround.

NB: sorry if that sounds harsh, I just have not enough time for it 😢

**Comment by theofidry** [comment]
In the meantime I still thinks this should be documented. Careful though as you are using the doc of master, which is for 3.x, instead of 2.x.

**Comment by damienalexandre** [comment]
I would like to report a similar issue with `v3.0.0-beta.4`.

```yml
A:
    a_template (template):
        reference: '<bothify("????##")>'
    a{1..500} (extends a_template): {}

B:
    b_template (template):
        slug: '<slug()>'
    b{1..20} (extends b_template):
        packages: '@a{1..100}'
```
```php
class A {
    private $reference;
}
class B {
    protected $slug;
    private $packages = [];
}
```

I'm using the NelmioAliceBundle Symfony bridge, and get this exception:

```                                                                               
  [Nelmio\Alice\Throwable\Exception\Generator\Resolver\ResolverNotFoundException]  
  No resolver found to resolve value "(choice) array (                             
    0 =>                                                                           
    Nelmio\Alice\Definition\Value\FixtureReferenceValue::__set_state(array(        
       'reference' => 'a1',                                     
    )),                                                                            
    1 =>                                                                           
    Nelmio\Alice\Definition\Value\FixtureReferenceValue::__set_state(array(        
       'reference' => 'a2',                                     
    )),                                                                            
    2 =>                                                                           
    Nelmio\Alice\Definition\Value\FixtureReferenceValue::__set_state(array(        
       'reference' => 'a3',                                     
    )),                                                                            
    3 =>                                                                           
    Nelmio\Alice\Definition\Value\FixtureReferenceValue::__set_state(array(        
       'reference' => 'a4',                                     
    )),                                                                            
```

I'm not familiar with Alice code base, but is there something I can do to help?

**Comment by theofidry** [comment]
@damienalexandre the error looks weird to me, I'll look into it. However you seem to misunderstand something:

```yaml
B:
    b_template (template):
        slug: '<slug()>'
    b{1..20} (extends b_template):
        packages: '@a{1..100}' # is equivalent to say pick one randomly in [@1, ..., @100]
```

c.f. https://github.com/nelmio/alice/blob/master/doc/advanced-guide.md#references

so I assume what you wanted to do was actually:

```yaml
B:
    b_template (template):
        slug: '<slug()>'
    b{1..20} (extends b_template):
        packages:
                - '@a{1..100}'
                - '@a{1..100}'
                - '@a{1..100}'
```

or better:

```yaml
B:
    b_template (template):
        slug: '<slug()>'
    b{1..20} (extends b_template):
        packages: '5x @a{1..100}' # eq as the above but ensures there's 5 items
```

If you want a random number of elements:

```yaml
B:
    b_template (template):
        slug: '<slug()>'
    b{1..20} (extends b_template):
        packages: '<numberBetween(0, 5)>x @a{1..100}'
```

And if you want to avoid duplicates, add the `(unique)` key to `packages`:

```yaml
B:
    b_template (template):
        slug: '<slug()>'
    b{1..20} (extends b_template):
        packages (unique): '<numberBetween(0, 5)>x @a{1..100}'
```

**Comment by damienalexandre** [comment]
Thanks for looking into it :)

I expect my syntax to return an array of 100 object "a", and set it to the "packages" variable, as described here: https://github.com/nelmio/alice/blob/master/doc/relations-handling.md#multiple-references
This is, in fact, the same syntax as the `pick one randomly in [@1, ..., @100]` so that's may be the issue.

What I'm trying to do is more like this:

```
B:
    b_template (template):
        slug: '<slug()>'
    b{1..20} (extends b_template):
        packages: 
            - '@a1' 
            - '@a2' 
            - '@a3' 
            ... 
            - '@a100'
```

I don't want anything random like `'<numberBetween(0, 5)>x @a{1..100}'` could do.

**Comment by theofidry** [comment]
Indeed the doc seems to disagree with me... I'll need to double check that to see if the behaviour is consistent between 2.x and 3.x.
